### PR TITLE
Feature: Allow stream custom maxsize per batch

### PR DESCRIPTION
**Pull request by simon28082**
## Problem
 
I'm trying to read the data using badger and use grpc for node sync, but I can't change the max cut size of stream because it's restricted hardcode

```go
// maxStreamSize is the maximum allowed size of a stream batch. This is a soft limit
// as a single list that is still over the limit will have to be sent as is since it
// cannot be split further. This limit prevents the framework from creating batches
// so big that sending them causes issues (e.g running into the max size gRPC limit).
var maxStreamSize = uint64(100 << 20) // 100MB
```
## Solution
 
Add a property to Stream to allow custom slice size

```go
type Stream struct {
	// Prefix to only iterate over certain range of keys. If set to nil (default), Stream would
	// iterate over the entire DB.
	Prefix []byte

	// Number of goroutines to use for iterating over key ranges. Defaults to 8.
	NumGo int

	// Badger would produce log entries in Infof to indicate the progress of Stream. LogPrefix can
	// be used to help differentiate them from other activities. Default is "Badger.Stream".
	LogPrefix string

	// ChooseKey is invoked each time a new key is encountered. Note that this is not called
	// on every version of the value, only the first encountered version (i.e. the highest version
	// of the value a key has). ChooseKey can be left nil to select all keys.
	//
	// Note: Calls to ChooseKey are concurrent.
	ChooseKey func(item *Item) bool

	// MaxSize is the maximum allowed size of a stream batch
	MaxSize uint64
}
```

### CODE REVIEW

**Comment by mangalaman93** [comment]
@simon28082 the linter is not happy. Could you check that please? and also pull in latest changes from the `main` branch?

**Comment by simon28082** [comment]
@mangalaman93 will do it

**Comment by simon28082** [comment]
@mangalaman93 I done it, please review

File: stream.go [hunk 1]
```diff
@@ -315,7 +323,7 @@ func (st *Stream) streamKVs(ctx context.Context) error {
 			// Send the batch immediately if it already exceeds the maximum allowed size.
 			// If the size of the batch exceeds maxStreamSize, break from the loop to
 			// avoid creating a batch that is so big that certain limits are reached.
-			if batch.LenNoPadding() > int(maxStreamSize) {
+			if batch.LenNoPadding() > uint64(st.MaxSize) {
```
Comment (mangalaman93) [inline stream.go:326]: I think what you should do here is this:
```
if uint64(batch.LenNoPadding()) > st.MaxSize {
```

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

File: stream.go (hunk 1 above)
Comment (simon28082) [reply stream.go]: modified

**Review by mangalaman93 — DISMISSED** [review]

File: stream.go [hunk 2]
```diff
@@ -315,7 +323,7 @@ func (st *Stream) streamKVs(ctx context.Context) error {
 			// Send the batch immediately if it already exceeds the maximum allowed size.
 			// If the size of the batch exceeds maxStreamSize, break from the loop to
 			// avoid creating a batch that is so big that certain limits are reached.
-			if batch.LenNoPadding() > int(maxStreamSize) {
```
Comment (mangalaman93) [inline stream.go:318]: could you try upgrading the action version in .github/workflows/ci-golang-lint.yml:29 to
```
uses: golangci/golangci-lint-action@v6.1.1
```

File: stream_test.go [hunk 3]
```diff
@@ -171,6 +171,68 @@ func TestStream(t *testing.T) {
 	require.NoError(t, db.Close())
 }
 
+
+func TestStreamMaxSize(t *testing.T) {
```
Comment (mangalaman93) [inline stream_test.go:175]: This file is not go formatted, please check

**Pull request merged**